### PR TITLE
Use `@ConfigProperties` instead of `@ConfigMappings`

### PR DESCRIPTION
**Pull request by snazy**
Reason: https://github.com/quarkusio/quarkus/issues/19990

Reverts: #1814

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/1966)

### CODE REVIEW

**Review by nastra — APPROVED** [review]

**Review by omarsmak — APPROVED** [review]

File: servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java [hunk 1]
```diff
@@ -53,7 +57,9 @@ public CompiledAuthorizationRules(QuarkusNessieAuthorizationConfig config) {
    * @return A map of compiled authorization rules
    */
   private Map<String, Script> compileAuthorizationRules() {
-    Map<String, String> rules = new HashMap<>(config.rules());
+
+    // Map<String, String> rules = new HashMap<>(config.rules());
```
Comment (dimas-b) [inline servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java:61]: nit: remove this comment?

File: servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java [hunk 2]
```diff
@@ -81,6 +87,26 @@ public CompiledAuthorizationRules(QuarkusNessieAuthorizationConfig config) {
...
+    return StreamSupport.stream(config.getPropertyNames().spliterator(), false)
+        .filter(name -> name.startsWith(prefix) && !name.equalsIgnoreCase(prefix))
+        .collect(
+            Collectors.toMap(
+                prop -> cleanPropertyName(prop.substring(prefix.length() + 1)),
+                prop -> config.getOptionalValue(prop, String.class).orElse("")));
+  }
+
+  private static String cleanPropertyName(String name) {
+    if (name.startsWith("\"") && name.endsWith("\"")) {
```
Comment (dimas-b) [inline servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java:104]: How can quotes occur in this context, given that `name` is a substring of a quarkus property?

**Review by dimas-b — DISMISSED** [review]

File: servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java (hunk 1 above)
Comment (snazy) [reply servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java:61]: It's the old code. Just left it so it's maybe easier to revert once the Quarkus issue is fixed.

File: servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusNessieAuthorizationConfig.java [hunk 3]
```diff
@@ -15,29 +15,28 @@
...
-import io.smallrye.config.ConfigMapping;
-import io.smallrye.config.WithDefault;
-import io.smallrye.config.WithName;
-import java.util.Map;
+import io.quarkus.arc.config.ConfigProperties;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
 
 /** Configuration for Nessie authorization settings. */
-@ConfigMapping(prefix = "nessie.server.authorization")
+@ConfigProperties(prefix = "nessie.server.authorization")
```
Comment (nastra) [inline servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusNessieAuthorizationConfig.java:22]: TBH we shouldn't roll back these changes. Then we also wouldn't need the workaround that was added in `CompiledAuthorizationRules`.

Also in fact I believe we only need to roll back the changes in `QuarkusServerConfig`. Everything else can use the new `ConfigMapping`

**Review by omarsmak — APPROVED** [review]

**Review by nastra — CHANGES_REQUESTED** [review]
I believe we only need to roll back the changes in `QuarkusServerConfig`. Everything else can use the new `ConfigMapping`

File: servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java (hunk 2 above)
Comment (dimas-b) [reply servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java:104]: TIL :)

File: servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusNessieAuthorizationConfig.java (hunk 3 above)
Comment (snazy) [reply servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusNessieAuthorizationConfig.java:22]: > only need to roll back the changes in QuarkusServerConfig

Everything that's `@Inject`ed via a constructor is affected by the Quarkus isssue.

Comment (nastra) [reply servers/quarkus-server/src/main/java/org/projectnessie/server/config/QuarkusNessieAuthorizationConfig.java:22]: as far as I remember, the problem was always the injection of `ServerConfig` into `NessieExceptionMapper`. The changes from https://github.com/projectnessie/nessie/pull/1967 work for me locally. Let's see if native tests pass with that minimal set of changes

**Pull request closed without merging**